src/common/history.c

```diff
@@ -1206,6 +1206,57 @@ GList *dt_history_duplicate(GList *hist)
   return result;
 }
 
+const char *dt_history_compute_hash(int32_t imgid)
+{
+  if(imgid == -1) return NULL;
+
+  GChecksum *checksum = g_checksum_new(G_CHECKSUM_MD5);
+
+  // get history
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "SELECT operation, op_params, enabled, blendop_params"
+                              " FROM main.history"
+                              " WHERE imgid = ?1"
+                              " ORDER BY num",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+
+  while(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    if(sqlite3_column_int(stmt, 2))
```

Comment:
Why not put this in the SQL statement above? Looping around all non enabled modules are time consuming for nothing.

Reply:
Right. I'll do.